### DISCUSSION TITLE
How should I get this to work in 1.4?

### DISCUSSION THREAD

**Discussion by the1gofer** [sqlalchemy/sqlalchemy#7841] [Usage Questions] [answered]
I had the following in 1.3 that worked, now that I've switched over to 1.4, I can't seem to figure out how to convert it. 

`session.query(URL.id, URL.url, URL.processed)`

I get the following:
<img width="1090" alt="Screen Shot 2022-03-22 at 9 20 48 AM" src="https://user-images.githubusercontent.com/2140724/159526213-4d1605ba-d41e-48e6-9a91-fe270e54b408.png">

I've seen some examples of joining from different tables, but how would I use this if I just want specific fields from one table?

**Comment by CaselIT** [comment]
Hi,

The straight forward conversion is
`session.execute(sa.select(URL.id, URL.url, URL.processed))`

If that does not work you need to provide further details, like how that model is definited

**Reply by the1gofer** [reply]
NameError: name 'sa' is not defined

What is sa supposed to be?   I don't recall seeing anything in the docs about it.  

Here is the model:
```
class URL(Base):
    """table to store urls"""
    __tablename__ = 'urls'

    # column
    id = Column(Integer, primary_key=True)
    _url = Column('url', String, nullable=False, unique=True)
    processed = Column(Boolean, default=False)
    domain = Column(Text, nullable=False)

    # relationships
    source = relationship("Source", back_populates="url", uselist=False)
    file = relationship("File", uselist=False, back_populates="url")

    def __init__(self, url: str, processed: bool = False):
        """URL instantiate creator"""
        self.url = url
        self.processed = processed
        self.domain = tldextract.extract(url).domain.lower()

    @property
    def url(self) -> str:
        return self._url

    @url.setter
    def url(self, url: str):
        if validators.url(url):
            self._url = url
        else:
            print ("raise InvalidURLError(url)")
    
    def __str__(self):
        return self.url

    def __repr__(self):
        return f"<url(url={self.url!r}, domain={self.domain!r}, processed={self.processed!r})>"
```

**Reply by CaselIT** [reply]
`sa` is a common abbreviation of sqlalchemy `import sqlalchemy as sa`

**Reply by CaselIT** [reply]
Looking at the model, the `url` is a proprty, so you need to select it using `_url` when accessing using the model class, otherwise it will not work correctly.
You may look into [hybrid_property](https://docs.sqlalchemy.org/en/14/orm/extensions/hybrid.html) or alternatively you could remove the property and use [validates](https://docs.sqlalchemy.org/en/14/orm/mapped_attributes.html?highlight=validates#sqlalchemy.orm.validates) to validate the value on set

**Reply by the1gofer** [reply]
thank you.  I'll look into those.  Idealy i'd like to have the model check it so I don't have to remember to do it every time I save it someplace else.

**Reply by the1gofer** [reply]
Looks like updating to @property to @hybrid_property did the trick. 

```
   @hybrid_property
    def url(self) -> str:
        return self._url
```

**Comment by the1gofer** [comment, marked as answer]
Looks like updating to https://github.com/Property to @hybrid_property did the trick.

```
   @hybrid_property
    def url(self) -> str:
        return self._url
```
